spec/middlewares/lead_provider_request_auditor_spec.rb

```diff
@@ -1,19 +1,15 @@
 # frozen_string_literal: true
 
-describe LeadProviderRequestAuditor do
-  let(:app) { proc { [200, {}, Time.zone.now.to_s] } }
-  let(:middleware) { described_class.new(app) }
-  let(:request) { Rack::MockRequest.new(middleware) }
-
+describe LeadProviderRequestAuditor, type: :request do
```

Comment:
Was the problem the middleware mock not being sufficient to test with? Shouldn't we have tests that replicate what raised and what happens in the event of a raise? This doesn't tell me what caused the raise; where the raise was handled; so hard to judge whether it fixes the issue.

Reply:
In fact, if I change the auditor implementation to the old version, I see
```
ActionDispatch::Http::Parameters::ParseError: no implicit conversion of nil into String
```
in all `requests/api/v1/participant_declarations_spec.rb` tests. Which is, I think, what we were seeing before.

Reply:
I don't think that's enough. Would expect a test (possibly with a stub) to make sure that what is passed in isn't changed when it is forwarded